server/src/server.ts

```diff
@@ -74,6 +78,127 @@ connection.onInitialize(
     },
 );
 
+connection.onInitialized((_params) => {
+    connection.workspace.getWorkspaceFolders().then((_value) => {
+        const workspacePath = url.fileURLToPath(_value[0].uri);
+        glob('**/project.kao', {
+            cwd: workspacePath,
+        })
+            .then((_matches) => {
+                for (const filePath of _matches) {
+                    if (openProjectFile(workspacePath, filePath)) {
+                        // There should be only one `project.kao` file. If we found a good candidate, stop the loop.
+                        return;
+                    }
+                }
+            })
+            .catch((_error) => console.error(_error));
+        /*
+         * List of the file-extensions of yseopml files that are never set by the user in `project.kao`-like files.
+         * This list is a subset of the file-extensions known by this extension as set in `client/package.json`.
+         */
+        const yseopmlExtensions = ['yclass', 'yobject', 'ycomplete'];
+        for (const extension of yseopmlExtensions) {
+            parseFilesWithExtension(workspacePath, extension);
+        }
+    });
+});
+
+/**
+ * Find all the files in the workspace that have the extension `extension`
+ * and open them as `TextDocument` objects. This will result in a parsing request for
+ * these files and have it known by the extension.
+ * This function excludes the results from `.generated-yml/`.
+ *
+ * @param extension The extension of the files to look for
+ */
+function parseFilesWithExtension(workspacePath: string, extension: string): void {
+    glob(`**/*.${extension}`, {
+        cwd: workspacePath,
+        ignore: '**/.generated-yml/**',
+    })
+        .then((_matches) => {
+            if (!_matches) {
+                return;
+            }
+            _matches.forEach((uri) => {
+                fsPromises
+                    .readFile(uri)
+                    .then((_doc) => {
+                        const doc = _doc.toString();
+                        parseFile(`file://${workspacePath}/${uri}`, doc);
+                    })
+                    .catch((_err) => {
+                        if (!!_err) {
+                            connection.console.error(`${_err}`);
+                            return;
+                        }
+                    });
+            });
+        })
+        .catch((_error) => console.error(_error));
+}
+
+const GENERATED_YML_DIR_REGEX = /(^|\/)\.generated-yml\//;
```

Comment:
Stranded const amidst functions and whatnot? :thinking:

Reply:
> Stranded const amidst functions and whatnot?

I only understand “functions” and “and” from this comment.

Reply:
C’mon! :pouting_woman: What I mean is that your code goes “weird stuff, then a function, then a constant, then a function again”.

Reply:
OK, I'll put related things together.